src/redux/reducers/product/productSlice.js

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import axios from "axios";
+// import url from "../../../constants/url";
+// import.meta.env.BASE_URL as;
+
+const initialState = {
+  loading: false,
+  products: [],
+  error: '',
+};
+const url = import.meta.env.VITE_BASE_URL;
```

Comment:
@sostene12 As I said in the `url.js` file, please put this into the constants so that other developers can access it.